src/libopensc/card-piv.c

```diff
@@ -49,18 +49,17 @@
 #endif
 #endif
 
+#include "internal.h"
+
 /* 800-73-4 SM and VCI need: ECC, SM and real OpenSSL >= 1.1 */
 #if defined(ENABLE_OPENSSL) && defined(ENABLE_SM) && !defined(OPENSSL_NO_EC) && !defined(LIBRESSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER >= 0x10100000L
+#include <openssl/cmac.h>
```

Comment:
This kind of check for prequisites should be done at configure time, when the user explicitly selects `--enable-piv-sm`. At compile time the user expects the feature to be included.